docs/docs/state-and-lifecycle.md

```diff
@@ -0,0 +1,504 @@
+  componentDidMount() {
+    this.intervalId = setInterval(this.tick, 1000);
+  }
+```
+
+Note how we save the interval ID right on `this`. While `this.props` is set up by React itself and `this.state` has a special meaning, you are free to add additional fields to the class manually if you need to store something that is not used for the visual output, such as the timeout ID.
+
+We will tear down the interval in the `componentWillUnmount()` lifecycle hook:
+
+```js{2}
+  componentDidMount() {
+    clearInterval(this.intervalId);
+  }
+```
+
+Finally, we will implement the `tick()` method that runs every second.
+
+It will use `this.setState()` to schedule updates to the component local state:
+
+```js{4,16-20}
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.tick = this.tick.bind(this);
+    this.state = {date: new Date()};
+  }
+
+  componentDidMount() {
+    this.intervalId = setInterval(this.tick, 1000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.intervalId);
+  }
+
+  tick() {
+    this.setState({
+      date: new Date()
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+
+const container = document.getElementById('root');
+ReactDOM.render(<Clock />, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/amqdNA?editors=0010)
+
+Now the clock ticks every second.
+
+## A Note on Binding Methods
+
+Note how we needed to bind `this.tick` in the constructor:
+
+```js{3}
+  constructor(props) {
+    super(props);
+    this.tick = this.tick.bind(this);
+    this.state = {date: new Date()};
+  }
+```
+
+In JavaScript, class methods are not [bound](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_objects/Function/bind) by default. If you forget to bind `this.tick` and pass it to `setInterval()`, `this` inside it will be `undefined`.
+
+This is not React-specific behavior, and is a part of [how functions work in JavaScript](https://www.smashingmagazine.com/2014/01/understanding-javascript-function-prototype-bind/).
+
+Generally, the rule of thumb is that if you refer to a method without `()` after it, such as `setTimeout(this.tick, 1000)`, you need to bind that method.
+
+You don't need to bind `render()` or any of the lifecycle methods.
+
+## Using the State Correctly
+
+There are three things you should know about `setState()`.
+
+### Do Not Modify State Directly
+
+For example, this will not work:
+
+```js
+// Wrong
+this.state.comment = 'Hello';
+```
+
+Instead, use `setState()`:
+
+```js
+// Correct
+this.setState({comment: 'Hello'});
+```
+
+### setState() Is Asynchronous
+
+React may batch multiple `setState()` calls into a single update for performance.
+
+Because `this.props` and `this.state` always represent the currently rendered state and props, you should not rely on their values when updating the state.
```

Comment:
Should we say that `setState` is asynchronous or that it may or may not be asynchronous? Does "asynchronous" cover both cases? It's possible that an update will update synchronously.